TombEngine/Game/Hud/PickupSummary.cpp

```diff
@@ -0,0 +1,223 @@
+		auto threshold = screenRes.ToVector2() * 0.1f;
+
+		// NOTE: Positions above screen can be ignored to account for high stacks.
+		if (checkAbove)
+		{
+			if (Position.y <= -threshold.y)
+				return true;
+		}
+
+		return (Position.x <= -threshold.x ||
+				Position.x >= (screenRes.x + threshold.x) ||
+				Position.y >= (screenRes.y + threshold.y));
+	}
+
+	void DisplayPickup::Update(bool isHead)
+	{
+		constexpr auto LIFE_BUFFER		   = 0.2f;
+		constexpr auto SCALE_MAX		   = 0.4f;
+		constexpr auto SCALE_MIN		   = 0.2f;
+		constexpr auto HIDE_VELOCITY_COEFF = 3 / 100.0f;
+		constexpr auto POSITION_LERP_ALPHA = 0.2f;
+		constexpr auto STRING_SCALAR_ALPHA = 0.75f;
+		const	  auto ROTATION			   = EulerAngles(0, ANGLE(3.0f), 0);
+
+		if (this->IsOffscreen(false))
+			return;
+
+		// Update position, scale, and opacity.
+		if (Life > 0.0f)
+		{
+			float totalDist = Vector2::Distance(Origin, Target);
+			float coveredDist = Vector2::Distance(Origin, Position);
+
+			// Handle edge case when stack shifts.
+			if (coveredDist > totalDist)
+			{
+				this->Origin = Position;
+				totalDist = Vector2::Distance(Origin, Target);
+				coveredDist = Vector2::Distance(Origin, Position);
+			}
+
+			float alpha = coveredDist / totalDist;
+
+			this->Position = Vector2::Lerp(Position, Target, POSITION_LERP_ALPHA);
+			this->Scale = std::max(Lerp(SCALE_MIN, SCALE_MAX, alpha), Scale);
+			this->Opacity = std::max(Lerp(0.0f, 1.0f, alpha), Opacity);
+		}
+
+		// Move offscreen.
+		if (isHead && Life <= 0.0f)
+		{
+			auto screenRes = g_Renderer.GetScreenResolution().ToVector2();
+			auto vel = Vector2(screenRes.x * HIDE_VELOCITY_COEFF, 0.0f);
+			this->Position += vel;
+		}
+
+		// Update orientation.
+		this->Orientation += ROTATION;
+
+		// Update string scale.
+		float alpha = Scale / SCALE_MAX;
+		this->StringScale = Lerp(0.0f, 1.0f, alpha) * (1.0f + StringScalar);
+		this->StringScalar *= STRING_SCALAR_ALPHA;
+
+		// Update life.
+		this->Life -= 1.0f;
+		if (!isHead)
+			this->Life = std::max(this->Life, round(LIFE_BUFFER * FPS));
+	}
+
+	void PickupSummaryController::AddDisplayPickup(GAME_OBJECT_ID objectID, const Vector3& pos)
+	{
+		constexpr auto LIFE_MAX			 = 2.5f;
+		constexpr auto STRING_SCALE_MAX	 = 2.0f;
+		constexpr auto STRING_SCALAR_MAX = 0.6f;
+
+		// TODO: Call this elsewhere. Maybe add PickUpObject() function to pickup.cpp.
+		PickedUpObject(objectID);
+
+		// Display pickup of same type exists; increment its count.
+		for (auto& pickup : this->DisplayPickups)
+		{
+			// Ignore already disappearing display pickups.
+			if (pickup.Life <= 0.0f)
+				continue;
+
+			if (pickup.ObjectID == objectID)
+			{
+				pickup.Count++;
+				pickup.Life = round(LIFE_MAX * FPS);
+				pickup.StringScale = STRING_SCALE_MAX;
+				pickup.StringScalar = STRING_SCALAR_MAX;
+				return;
+			}
+		}
+
+		// Create new display pickup.
+		auto& pickup = GetNewDisplayPickup();
+
+		pickup.ObjectID = objectID;
+		pickup.Count = 1;
+		pickup.Position =
```

Comment:
oopsie

Reply:
Nope, this is intentional. :P `Position` is assigned to the line below.

Reply:
oh okay, first time seeing this aha